src/stratigraphy/util/predictions.py

```diff
@@ -0,0 +1,172 @@
+"""This module contains classes for predictions."""
+
+import fitz
+import Levenshtein
+
+from stratigraphy.benchmark.ground_truth import GroundTruthForFile
+from stratigraphy.util.depthcolumnentry import DepthColumnEntry
+from stratigraphy.util.line import TextLine, TextWord
+from stratigraphy.util.util import parse_text
+
+
+class MaterialDescriptionPrediction:
+    """A class to represent a material description prediction.
+
+    TODO: Check if this class can be replaced.
+    """
+
+    def __init__(self, text: str, rect: list, lines: list):
+        self.text = text
+        self.rect = fitz.Rect(rect)
+        self.lines = [TextLine([TextWord(**line)]) for line in lines]
+
+
+class DepthIntervalPrediction:
+    """A class to represent a depth interval prediction.
+
+    TODO: Could be replaced by BoundaryInterval from interval.py. Then we could use line_anchor and background_rect.
+    """
+
+    def __init__(self, start: dict, end: dict):
+        self.start = (
+            DepthColumnEntry(value=start["value"], rect=fitz.Rect(start["rect"])) if start is not None else None
+        )
+        self.end = DepthColumnEntry(value=end["value"], rect=fitz.Rect(end["rect"])) if end is not None else None
+
+
+class LayerPrediction:
+    """A class to represent predictions for a single layer."""
+
+    def __init__(self, material_description: MaterialDescriptionPrediction, depth_interval: DepthIntervalPrediction):
+        self.material_description = material_description
+        self.depth_interval = depth_interval
+        self.material_is_correct = None
+        self.depth_interval_is_correct = None
+
+
+class PagePredictions:
+    """A class to represent predictions for a single page."""
+
+    def __init__(
+        self, layers: list[LayerPrediction], page_number: int, depths_materials_columns_pairs: list[dict] = None
+    ):
+        self.layers = layers
+        self.page_number = page_number
+        self.depths_materials_columns_pairs = depths_materials_columns_pairs
+
+
+class FilePredictions:
+    """A class to represent predictions for a single file."""
+
+    def __init__(self, pages: list[PagePredictions] = None, file_name: str = None):
```

Comment:
How about:
- Make the parameters required here (no more default None values)
- Make the create_from_json method static, and have it return a FilePredictions instance
- Initialize the FilePredictions instance by directly calling FilePredictions.create_from_json(file_predictions, file_name)
- _check_if_initialized method can be removed?